madara/crates/client/db/src/lib.rs

```diff
@@ -318,6 +326,96 @@ impl MadaraBackend<RocksDBStorage> {
         let db = RocksDBStorage::open(&db_path, rocksdb_config).context("Opening rocksdb storage")?;
         Ok(Arc::new(Self::new_and_init(db, chain_config, config)?))
     }
+
+    /// Rollback the database to a specific block number
+    /// This removes all blocks after the specified block number
+    /// and reverts the state to that point
+    pub fn rollback_to_block(&self, target_block_n: u64) -> Result<()> {
+        tracing::warn!("⏮️ Rolling back database to block #{}", target_block_n);
+
+        // Get the current latest block
+        let current_block = self.latest_confirmed_block_n();
+        
+        if let Some(current) = current_block {
+            if target_block_n >= current {
+                tracing::info!("Target block {} is >= current block {}, nothing to rollback", target_block_n, current);
+                return Ok(());
+            }
+        } else {
+            tracing::warn!("No blocks found in database, cannot rollback");
+            return Ok(());
+        }
+
+        let current_block = current_block.unwrap();
```

Comment:
we won't  need to unwrap the current_block again!
```suggestion
       let Some(current_block) = current_block else {
            tracing::warn!("No blocks found in database, cannot rollback");
            return Ok(());
       };

       if target_block_n >= current_block {
           tracing::info!(
                  "Target block {} is >= current block {}, nothing to rollback",
                    target_block_n, current_block
            );
           return Ok(());
      }
```

Reply:
reformatted